### PR TITLE
New blog: Small PVs and Kadalu Storage

### PR DESCRIPTION
**Pull request by aravindavk**
Signed-off-by: Aravinda Vishwanathapura <mail@aravindavk.in>

### CODE REVIEW

File: _posts/2020-02-06-small-pvs-and-kadalu-storage.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
+---
+title: "Small PVs and Kadalu Storage"
+author: aravindavk
+layout: post
+description: "Kadalu Storage can provide the smallest Persistent Volume as low as 20MiB for the applications running in Kubernetes(k8s). Small PVs are attractive for many use cases like providing home dir for each container user or storing config files of applications etc."
```
Comment (sankarshanmukhopadhyay) [inline _posts/2020-02-06-small-pvs-and-kadalu-storage.md]: Should it be be Kadalu Storage can provide or, Kadalu Storage can provision?

File: _posts/2020-02-06-small-pvs-and-kadalu-storage.md [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    persistentVolumeClaim:
+      claimName: pv1
+  restartPolicy: OnFailure
+```
+
+**Note**: The container image used in the above example is non-existent. It is used for illustrative purposes only.
+
+## Minimize the Risk
+
+What if Pod terminates abruptly or node goes down without calling Pre to stop hook script? Let's create a background job that syncs data from `$HOME` to `/srcdata` at regular intervals. For example, if the background job runs once every 10 seconds, then the risk of losing data is reduced to 10 seconds(plus sync time).
```
Comment (sankarshanmukhopadhyay) [inline _posts/2020-02-06-small-pvs-and-kadalu-storage.md]: What happens in cases where the Pod terminates abruptly or, the node goes down without being able to call Pre to stop the hook script?

File: _posts/2020-02-06-small-pvs-and-kadalu-storage.md [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+**Note**: This approach is not suitable for large volumes, but a similar technique can be adapted for medium and large volumes if Inotify(or similar tools) is available for containers.
```
Comment (sankarshanmukhopadhyay) [inline _posts/2020-02-06-small-pvs-and-kadalu-storage.md]: I think we should add a line to emphasize that this approach will still have data loss. The specific use case creates a trade off that losing data for such minimal time duration is considered to be acceptable.

**Review by sankarshanmukhopadhyay — CHANGES_REQUESTED** [review]

**Review by amarts — APPROVED** [review]

**Comment by aravindavk** [comment]
Do not merge now, we can plan to publish next week

**Review by sankarshanmukhopadhyay — APPROVED** [review]

**Pull request merged**